pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -385,9 +399,30 @@ EOF_VIA_SSH`, data))
 	return c.runSSH(`chmod +x /root/post-install.sh`)
 }
 
-// GetRunningInstallImageProcesses returns the running installimage processes. Output.StdOut is empty if no processes are running.
-func (c *sshClient) GetRunningInstallImageProcesses() Output {
-	return c.runSSH(`ps aux| grep installimage | grep -v grep; true`)
+// GetInstallImageState returns the running installimage processes. Output.StdOut is empty if no processes are running.
+// Explanation of return values:
+//
+//	false, false, nil --> install-image has not started yet.
+//	true,  false, nil --> install-image is still running
+//	false, true,  nil --> install-image has finished.
+//	true,  true,  nil --> this should not happen.
+//	*,     *,     err --> ssh to machine failed.
+func (c *sshClient) GetInstallImageState() (running bool, finished bool, err error) {
```

Comment:
I think it will get more complicated. I think it is currently simple and easy to understand. If you look at the place where the method gets used, then I think it is easy to understand.

Reply:
Sorry, but I disagree :D You have two booleans two define the three states. Right now it is not easily understandable and we need the explanation that you put above the function to even understand the return values. A proper type will make things understandable out of the box.

Reply:
In 6c1f28fd I added the new type InstallImageState.